react/week4/gitusers-app/src/components/GetRipos.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState, useContext } from 'react';
+import Modal from './Modal';
+import { HandlerContext } from './GitBase';
+
+export function GetRipos() {
+  const { userName, ripos, setRipos } = useContext(HandlerContext);
+  const [errorMessage, setErrorMessage] = useState('');
+  useEffect(() => {
+    (async () => {
+      try {
+        const result = await fetch(
+          `https://api.github.com/users/${userName}/repos`
+        );
+        if (result.status !== 200) {
+          throw new Error('fail to connect api for ripos');
+        }
+        const content = await result.json();
+        const riposList = content;
+        setRipos(prev => riposList.map(ripo => ripo.name));
+      } catch (error) {
+        setErrorMessage(prev => error.message);
+      }
+    })();
+  }, [userName]);
+
+  const riposLi = ripos.map((item, index) => <div key={index}>{item}</div>);
+
+  return (
+    <div>
+      <Modal>
+        {!riposLi.length && !errorMessage
+          ? 'No ripo'
+          : errorMessage
```

Comment:
bit confuse here , what exactly you want to do it , so what I understand from this condition is if there is list if empty and no error message then show 'N repo' otherwise error message but if there is not error message then again show error message else show repo. its not clear as you already checking error message condition on first check, no need to check again.